pydatastructs/trees/heaps.py

```diff
@@ -0,0 +1,223 @@
+from pydatastructs.utils.misc_util import _check_type
+
+NoneType = type(None)
+
+__all__ = [
+        'BinaryHeap'
+]
+
+class BinaryHeap():
+    """
+    Respresentation of BinaryHeap data structure
+
+    Parameters
+    ==========
+
+    array : list

```

Comment:
I would suggest to use, `OneDimensionalArray`.

Reply:
If we use ODA then we have to give our heap size/max_size also. So that we can insert an element in heap.
Implementation -
Parameters
=========
elements : ODA
     Default none.
     ODA having initial elements in Heap.
max_size : int
     Maximum size of Heap.

Note
=========
If ODA and max_size passed a new ODA of max_size is formed to hold elements of Heap and passes elements will be copied.

 Else passed ODA will determine the size of our heap and we can't insert an element.

Or you want me just pass an ODA and can use the python list to store the elements of the heap.

Reply:
@czgdp1807 What changes you want me to do about this in th code.